### PR TITLE
NO-ISSUE: use ocp builder instead of openshift/release golang image

### PR DESCRIPTION
**Pull request by rccrdpccl**
/cc @tsorya

### CODE REVIEW

File: Dockerfile.assisted_installer_agent-build [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM registry.ci.openshift.org/openshift/release:golang-1.18
```
[the remark this answers is not in the document]
Comment (rccrdpccl) [reply Dockerfile.assisted_installer_agent-build:1]: we need to install differently docker-compose, but yes, that was the idea

**Comment by rccrdpccl** [comment]
@gamli75 @filanov This is changing upstream image to private: the benefit is that we would have upstream build with this flag thus enable to install clusters in FIPS mode, the disadvantage is that to build this project you need RH credentials for pulling the base image.
Is it worth pursuing this in upstream or shall we just do it in downstream images? Any thoughts?

**Comment by gamli75** [comment]
what about the opened PR: https://github.com/openshift/assisted-installer-agent/pull/610/files? @danmanor

**Comment by gamli75** [comment]
changing the upstream image to a private one can be problematic because Nokia are using podman with upstream images 
@tsorya ?

**Comment by rccrdpccl** [comment]
> what about the opened PR: https://github.com/openshift/assisted-installer-agent/pull/610/files? @danmanor

Something like that would be the alternative, yes. Initially I thought we are already using private images in other repos and I tried the complete approach (building with FIPS flag upstream), but I am having second thoughts now.

**Comment by rccrdpccl** [comment]
Ok, thanks for the info - this definitely needs further discussion if we want to go ahead, I'll close it for now (and follow the regular upgrade path)

**Pull request closed without merging**